devices/simulator.go

```diff
@@ -403,7 +467,121 @@ func (s Simulator) Info() (*FullDeviceInfo, error) {
 	}, nil
 }
 
-func (s Simulator) StartScreenCapture(format string, quality int, scale float64, callback func([]byte) bool) error {
-	mjpegClient := mjpeg.NewWdaMjpegClient("http://localhost:9100")
+func (s *SimulatorDevice) StartScreenCapture(format string, quality int, scale float64, callback func([]byte) bool) error {
+	mjpegPort, err := s.getWdaMjpegPort()
+	if err != nil {
+		return fmt.Errorf("failed to get MJPEG port: %w", err)
+	}
+
+	mjpegClient := mjpeg.NewWdaMjpegClient(fmt.Sprintf("http://localhost:%d", mjpegPort))
 	return mjpegClient.StartScreenCapture(format, callback)
 }
+
+func findWdaProcessForDevice(deviceUDID string) (int, string, error) {
+	cmd := exec.Command("/bin/ps", "-o", "pid,command", "-E", "-ww", "-e")
+	output, err := cmd.Output()
+	if err != nil {
+		return 0, "", fmt.Errorf("failed to run ps command: %w", err)
+	}
+
+	lines := strings.Split(string(output), "\n")
+	devicePath := fmt.Sprintf("/Library/Developer/CoreSimulator/Devices/%s", deviceUDID)
+
+	for _, line := range lines {
+		if strings.Contains(line, devicePath) && strings.Contains(line, "WebDriverAgentRunner-Runner") {
+			// Find the first space to separate PID from the rest
+			spaceIndex := strings.Index(line, " ")
+			if spaceIndex == -1 {
+				continue
+			}
+
+			pidStr := strings.TrimSpace(line[:spaceIndex])
+			pid, err := strconv.Atoi(pidStr)
+			if err != nil {
+				continue
+			}
+
+			// The rest of the line contains command and environment
+			processInfo := line[spaceIndex+1:]
+			return pid, processInfo, nil
+		}
+	}
+
+	return 0, "", fmt.Errorf("WebDriverAgent process not found for device %s", deviceUDID)
+}
+
+func extractEnvValue(output, envVar string) (string, error) {
+	// Look for " ENVVAR=" pattern (space + envvar + equals)
+	pattern := " " + envVar + "="
+	pos := strings.Index(output, pattern)
+	if pos == -1 {
+		// Also check if it's at the beginning of the line
+		pattern = envVar + "="
```

Comment:
not going to happen ever, because the string starts with the command itself, not with a " "